diopi_test/csrc/litert.cpp

```diff
@@ -150,10 +150,7 @@ diopiTensor::diopiTensor(const diopiSize_t* shape, const diopiSize_t* stride, di
     } else {
         storage_ = std::make_shared<Storage>(device_malloc, device_free, nbytes);
         if (src != nullptr) {
-            diopiStreamHandle_t stream;
-            diopiGetStream(context, &stream);
-            device_memcpy_h2d_async(stream, storage_->data(), src, nbytes);
-            device_synchronize_stream(stream);
+            diopiTensorCopyFromBuffer(context, src, this);
         }
     }
     context_ = context;
```

Comment:
move this line ahead? for example, before line 148
it may use the context_ passed along with tensor somewhere else and it fails if context_ is null

Reply:
抱歉，没太理解什么意思，能用中文说明一下吗？ @Aaron20000101

Reply:
建议把这一行移动到148行前。这样的话，如果在diopiTensorCopyFromBuffer里，有代码想通过tensor来方位context的话，不会出错。

Reply:
我可能缺少您那边调用的语境，所以不太理解“在diopiTensorCopyFromBuffer里，再调用其它代码，想通过tensor来访问context”是什么场景，您能否给一个例子呢？

Reply:
举个例子。
在diopiTensorCopyFromBuffer的（厂商开发的）实现体**里**，执行完数据转换和复制后，想对设备上的数据进行格式验证，所以调用Tensor::buffer() 把数据再复制回到CPU，这时在litert.hpp 的111行就会报错，因为context_为空。

这个修改建议是希望能在执行diopiTensorCopyFromBuffer前，尽可能把所有已知数据准备好，这样安全性比较高。

Reply:
哦哦我理解了，我稍后再提一个pr修复这个问题

Reply:
Done @Aaron20000101